src/adaptors.rs

```diff
@@ -70,6 +70,84 @@ impl<I, J> Iterator for Interleave<I, J> where
     }
 }
 
+/// An iterator adaptor that alternates elements from the two iterators until
+/// one of them runs out.
+///
+/// This iterator is *fused*.
+///
+/// See [*.interleave_shortest()*](trait.Itertools.html#method.interleave_shortest)
+/// for more information.
+#[derive(Clone)]
+pub struct InterleaveShortest<I, J> where
+    I: Iterator,
+    J: Iterator<Item=I::Item>,
+{
+    it0: Fuse<I>,
+    it1: Fuse<J>,
+    phase: bool, // false ==> it0, true ==> it1
+}
+
+impl<I, J> InterleaveShortest<I, J> where
+    I: Iterator,
+    J: Iterator<Item=I::Item>,
+{
+    /// Create a new **InterleaveShortest** iterator.
+    pub fn new(a: I, b: J) -> InterleaveShortest<I, J> {
+        InterleaveShortest {
+            it0: a.fuse(),
+            it1: b.fuse(),
+            phase: false,
+        }
+    }
+}
+
+impl<I, J> Iterator for InterleaveShortest<I, J> where
+    I: Iterator,
+    J: Iterator<Item=I::Item>,
+{
+    type Item = I::Item;
+
+    #[inline]
+    fn next(&mut self) -> Option<I::Item> {
+        match self.phase {
```

Comment:
You can, but I really didn't want to do that because then you need to match on `!phase`, which is weird.  If you want to cut the repetition out, I suppose I can bind the result of the match to a local `r`, then flip `phase` *after* the match, then return `r`... but the code would be the same length *anyway*.

I dunno; might just be my penchant for state machines.  :P